src/cli.ts

```diff
@@ -35,38 +44,50 @@ const options = yargs()
       describe: "output file",
       type: "string",
     },
+    chars: {
+      alias: "c",
+      description: "Use char length as objective",
+      type: "boolean",
+    },
   })
   .parseSync(process.argv.slice(2));
 
-const lang = languageTable[options.lang];
+const langs =
+  options.lang === "all" ? languages : [languageTable[options.lang]];
 const code = fs.readFileSync(options.input, { encoding: "utf-8" });
 const prog = parse(code);
-try {
-  const result = applyLanguage(lang, prog, searchOptions("full", "bytes"));
-  if (options.output !== undefined) {
-    fs.mkdirSync(path.dirname(options.output), { recursive: true });
-    fs.writeFileSync(options.output, result);
-  } else {
-    console.log(result);
-  }
-} catch (e) {
-  if (e instanceof PolygolfError) {
-    console.log(e.message);
-    if (e.source != null) {
-      const startLine = e.source.line === 0 ? 0 : e.source.line - 2;
-      console.log(
-        code
-          .split(/\r?\n/)
-          .slice(startLine, e.source.line)
-          .map((x, i) => `${startLine + i + 1}`.padStart(3, " ") + " " + x)
-          .join("\n") +
-          "\n" +
-          " ".repeat(e.source.column + 3) +
-          "^"
-      );
+for (const lang of langs) {
+  try {
+    const result = applyLanguage(
+      lang,
+      prog,
+      searchOptions("full", options.chars === true ? "chars" : "bytes")
+    );
+    if (options.output !== undefined) {
+      fs.mkdirSync(path.dirname(options.output), { recursive: true });
+      fs.writeFileSync(options.output + "." + lang.extension, result);
+    } else {
+      console.log(result);
```

Comment:
The output generated on the command line when compiling `polygolf -i XXX.polygolf -l all` could benefit from spacing between and/or labeling of the various solutions it outputs.